src/metabase/api/database.clj

```diff
@@ -983,10 +984,20 @@
   ;; just wrap this in a future so it happens async
   (let [db (api/write-check (t2/select-one Database :id id))]
     (events/publish-event! :event/database-manual-sync {:object db :user-id api/*current-user-id*})
-    (future
-      (sync-metadata/sync-db-metadata! db)
-      (analyze/analyze-db! db)))
-  {:status :ok})
+    (if-let [ex (try (do
+                       ;; it's okay to allow testing H2 connections during sync. We only want to disallow you from testing them for the
+                       ;; purposes of creating a new H2 database.
+                       (binding [h2/*allow-testing-h2-connections* true]
```

Comment:
is this binding only used in tests?

if so you can bind this directly when you call the API because `mt/user-http-request` now makes an in-thread request.

Reply:
No, "testing" in `h2/*allow-testing-h2-connections*` is not testing as in tests, it is testing as in checking the connection. This binding is used in many places outside tests  e.g. https://github.com/metabase/metabase/blob/cb2ba50f3d4a7b7fc9f1634b27681514b561d940/src/metabase/api/table.clj#L84